core/testcontainers/core/reaper.py

```diff
@@ -0,0 +1,71 @@
+from os import environ
+from socket import socket
+from typing import TYPE_CHECKING, Optional
+
+
+from .utils import setup_logger
+from .images import REAPER_IMAGE
+from .waiting_utils import wait_for_logs
+from .labels import LABEL_SESSION_ID, SESSION_ID
+
+if TYPE_CHECKING:
+    from .container import DockerContainer
+
+
+logger = setup_logger(__name__)
+
+
+class Reaper:
```

Comment:
The way things are currently organised, there will be problems with circular imports as `DockerContainer` depends on creating a `Reaper` as part of it's basic functionality, and `reaper.py` therefore cannot import anything from `container.py` without issues. This is also why there is a lazy import from `container` in the `Reaper._create_instance` method, to avoid this problem. 

The `Reaper` instance is not accessed through a DockerContainer instance, and is only available through its singleton module-level variable, so we won't be having cases of `postgres_container._container._container` behaviour or similar. The `_container` field on the `Reaper` instance is also meant to be a class member in order to access it as a singleton, not an instance member. If we were to make it into an instance field we would need to access it as `Reaper._instance._container`, which is not very clean IMO